crates/solidity/testing/utils/src/cst_snapshots/test_nodes.rs

```diff
@@ -25,60 +23,44 @@ pub struct TestNode {
 impl TestNode {
     pub fn from_cst(node: &Rc<cst::Node>) -> Self {
         return match node.as_ref() {
-            cst::Node::Rule { kind, children } => {
+            cst::Node::Rule {
+                kind,
+                span,
+                children,
+            } => {
                 let children = children
                     .iter()
                     .filter(|child| !Self::is_whitespace(child))
                     .map(|child| Self::from_cst(child))
                     .collect();
 
-                let range = Self::calculate_range(&children);
-
                 Self {
                     kind: TestNodeKind::Rule(*kind),
-                    range,
+                    range: Some(span.clone()),
                     children,
                 }
             }
-            cst::Node::Group { children } => {
-                let children = children
-                    .iter()
-                    .filter(|child| !Self::is_whitespace(child))
-                    .map(|child| Self::from_cst(child))
-                    .collect();
-
-                let range = Self::calculate_range(&children);
-
-                Self {
-                    kind: TestNodeKind::Group,
-                    range,
-                    children,
-                }
-            }
-            cst::Node::Token {
-                kind,
-                lex_node,
-                trivia,
-            } => Self::from_token(kind, lex_node, trivia),
+            cst::Node::Token { kind, span, trivia } => Self::from_token(kind, span, trivia),
         };
     }
 
     fn from_token(
```

Comment:
Should we now remove `calculate_range()` since the public API does that? This would also make sure the new API is covered by tests.

Reply:
yes, although token range does not include the trivia attached to the node.

Reply:
Aha! perhaps we can rename it to something like `calculate_range_with_trivia()` to distinguish it from `node.range()`.

Reply:
agreed

Reply:
added `range_with_trivia` to node(s}

Reply:
To the public API? even better!
In that case, is there any other use case for `TestNode::calculate_range()`? should we remove it?

Reply:
done